crates/code_generation/src/run.rs

```diff
@@ -3,210 +3,43 @@ use std::iter::repeat;
-                        quote! {
-                            let (#producer_identifier, #consumer_identifier) = framework::future_queue();
-                        }
-                    })
-                    .collect::<Vec<_>>(),
-                )
-            } else {
-                None
-            }
-        })
-        .flatten()
-        .collect();
-    let cycler_starts: Vec<_> = cyclers
-        .iter()
-        .flat_map(|cycler| {
-            cycler.get_cycler_instances().modules_to_instances[cycler.get_cycler_module_name()]
-                .iter()
-                .map(|cycler_instance| {
-                    let cycler_instance_snake_case = cycler_instance.to_case(Case::Snake);
-                    let cycler_variable_identifier =
-                        format_ident!("{}_cycler", cycler_instance_snake_case);
-                    let cycler_handle_identifier =
-                        format_ident!("{}_handle", cycler_instance_snake_case);
-                    let error_message = format!("failed to start cycler `{cycler_instance}`");
-                    quote! {
-                        let #cycler_handle_identifier = #cycler_variable_identifier
-                            .start(keep_running.clone())
-                            .wrap_err(#error_message)?;
-                    }
-                })
-                .collect::<Vec<_>>()
-        })
-        .collect();
-    let cycler_joins: Vec<_> = cyclers
-        .iter()
-        .flat_map(|cycler| {
-            cycler.get_cycler_instances().modules_to_instances[cycler.get_cycler_module_name()]
-                .iter()
-                .map(|cycler_instance| {
-                    let cycler_instance_snake_case = cycler_instance.to_case(Case::Snake);
-                    let cycler_handle_identifier =
-                        format_ident!("{}_handle", cycler_instance_snake_case);
-                    quote! {
-                        match #cycler_handle_identifier.join() {
-                            Ok(Err(error)) => {
-                                encountered_error = true;
-                                println!("{error:?}");
-                            },
-                            Err(error) => {
-                                encountered_error = true;
-                                println!("{error:?}");
-                            },
-                            _ => {},
-                        }
-                    }
-                })
-                .collect::<Vec<_>>()
-        })
-        .collect();
     quote! {
-        #[allow(unused_imports, unused_variables)]
-        pub fn run<Interface>(
-            hardware_interface: std::sync::Arc<Interface>,
+        #[allow(dead_code, unused_mut, unused_imports, unused_variables, clippy::too_many_arguments, clippy::redundant_clone, clippy::needless_question_mark)]
+        pub(crate) fn run(
```

Comment:
Should this even be public?

Reply:
It has to be, since the run function is what is being called in the binary main functions.